src/test/unit/flight_failsafe_unittest.cc

```diff
@@ -197,7 +197,11 @@ TEST(FlightFailsafeTest, TestFailsafeDetectsRxLossAndStartsLanding)
         failsafeUpdateState();
 
         // then
-        EXPECT_EQ(FAILSAFE_RX_LOSS_DETECTED, failsafePhase());
+        if (i < MAX_COUNTER_VALUE_WHEN_RX_IS_RECEIVED_AFTER_RX_CYCLE-1) {
```

Comment:
Logic in tests is not a good idea, unless the logic in the test code is tested elsewhere (e.g. test library).